### ISSUE TITLE
Onchain balance inaccurate

### ISSUE THREAD

**Issue by tdickman** [alexbosworth/balanceofsatoshis#18]
The value returned by `bos balance --onchain` appears to be `total_balance + unconfirmed_balance` rather than `confirmed_balance + unconfirmed_balance`.

**Comment by alexbosworth** [comment]
Ah I think it is confirmed_balance + (confirmed and unconfirmed balance), will fix

**Comment by alexbosworth** [comment]
It seems like the issue is that LND reports unconfirmed balance as either 0 when there is no unconfirmed, or the full balance inclusive of unconfirmed when there is unconfirmed but the code is assuming that the unconfirmed is always the total of unconfirmed funds

**Comment by alexbosworth** [comment]
Hmm I can't repro this though

**Comment by tdickman** [comment]
More details:

```
$ bos balance --onchain
439445736
$ bos balance --offchain
437984169
$ lncli walletbalance
{
    "total_balance": "274517732",
    "confirmed_balance": "109589728",
    "unconfirmed_balance": "164928004"
}

$ ksh lnd lncli channelbalance
{
    "balance": "437984169",
    "pending_open_balance": "0"
}
```

I am running lnd 0.9.0 and bos 5.28.2.

**Comment by alexbosworth** [comment]
I just tried it on my testnet node to sanity check:

1. Starting on-chain balance `bos balance --onchain`: `17mm`
1. Send `1mm` on-chain
1. Check on-chain balance `bos balance --onchain`: `18mm`
1. Check on-chain confirmed balance: `bos balance --onchain --confirmed`: `17mm`
1. Check total balance with no --onchain flag: `32mm`

This seems to suggest that `--onchain` does sum up the confirmed and unconfirmed components as expected

**Comment by alexbosworth** [comment]
Oh I think you are not looking at the pending channel onchain

**Comment by alexbosworth** [comment]
Offline chat: potentially the `total_limbo_balance` in pending channels that reports pending close channel balance amount is duplicative with the walletbalance unconfirmed balance, will look into that

**Comment by alexbosworth** [comment]
This requires a more complex fix because some of the on-chain balance reported by LND is duplicative and sometimes is non-duplicative. It should only temporarily be wrong until a confirmation but I will try and address it as part of broad changes needed to support 0.10.0

**Comment by alexbosworth** [comment]
The way I'm thinking about resolving this in the short term is to remove the entire limbo balance from the output because some of it is duplicative, even though some of it is non-duplicative

Then in the future I can revisit it and figure out how to include the non-duplicative component to get the full picture of the on-chain balance

**Comment by alexbosworth** [comment]
Should be resolved in 5.29.1
